RUFAS/EEE/tractor_implement.py

```diff
@@ -0,0 +1,192 @@
+            self.operation_type = OperationType.MOWING
+        for data_entry in dataset:
+            if (
+                data_entry.get("Crop Type or Tillage Implement").lower() in [crop_type_or_tillage_implement, "none"]
+                and data_entry.get("Tractor Size").lower() == self.tractor_size.value.lower()
+                and data_entry.get("Operation").lower() == self.operation_type.value.lower()
+            ):
+                self.implement_name: str = data_entry["Tractor Implement"]
+                self.A: float = data_entry["Tractor A (unitless)"]
+                self.B: float = data_entry["Tractor B (unitless)"]
+                self.C: float = data_entry["Tractor C (unitless)"]
+                self.E: float = data_entry["Tractor E (unitless)"]
+                self.F: float = data_entry["Tractor F (unitless)"]
+                self.G: float = data_entry["Tractor G (unitless)"]
+                self.width_m: float = data_entry["Tractor Implement Width (m)"]
+                self.mass_kg: float = data_entry["Tractor Implement Mass (kg)"]
+                self.throughput: float = data_entry["Max Throughput (tons dm/hour)"]
+                if self.operation_type in [
+                    OperationType.TILLING,
+                    OperationType.FERTILIZER_APPLICATION_BELOW_SURFACE,
+                    OperationType.FERTILIZER_APPLICATION_SURFACE,
+                    OperationType.LIQUID_MANURE_APPLICATION_BELOW_SURFACE,
+                    OperationType.LIQUID_MANURE_APPLICATION_SURFACE,
+                ]:
+                    self.depth_cm: float = application_depth * GeneralConstants.MM_TO_CM
+                else:
+                    self.depth_cm = data_entry["Depth"]
+                self.is_depth_relevant: bool = data_entry["is depth relevant"]
+                break
+
+    def field_capacity_ha_per_hr(
+        self, crop_yield_ton_per_ha: float | None, application_mass: float | None = None
+    ) -> float:
+        """
+        Calculates the Field Capacity for a specific crop, field operation and tractor implement.
+        Implements Helper Functions 418a and 418b in EEE Functions file.
+        """
+        if self.operation_type == OperationType.COLLECTION and self.harvest_type != HarvestOperation.KILL_ONLY:  # 418b
+            return (self.throughput / crop_yield_ton_per_ha) * self.field_efficiency
+        elif self.operation_type in [
+            OperationType.LIQUID_MANURE_APPLICATION_BELOW_SURFACE,
+            OperationType.LIQUID_MANURE_APPLICATION_SURFACE,
+        ]:      # 418c
+            return (self.throughput / application_mass) * self.field_efficiency
+
+        return (
+            self.field_speed_km_per_hr
+            * GeneralConstants.KM_TO_M
+            * self.width_m
+            * self.field_efficiency
+            * GeneralConstants.SQUARE_METERS_TO_HECTARES
+        )       # 418a
```

Comment:
```suggestion
        ) 
```
These should either be removed or added to the docstrings.